pkg/service/auth_pipeline.go

```diff
@@ -498,19 +475,63 @@ func (pipeline *AuthPipeline) Evaluate() auth.AuthResult {
 			authResult <- result
 		}
 
-		metrics.ReportTimedMetric(authServerAuthConfigDurationMetric, evaluateFunc, pipeline.metricLabels()...)
+		metrics.ReportTimedMetric(metrics.Registry.GetAuthServerAuthConfigDurationMetric(), evaluateFunc, pipeline.metricLabels())
 	}()
 
 	return <-authResult
 }
 
 func (pipeline *AuthPipeline) reportStatusMetric(rpcStatusCode rpc.Code) {
-	metrics.ReportMetricWithStatus(authServerAuthConfigResponseStatusMetric, rpc.Code_name[int32(rpcStatusCode)], pipeline.metricLabels()...)
+	metrics.ReportMetricWithStatus(metrics.Registry.GetAuthServerAuthConfigResponseStatusMetric(), rpc.Code_name[int32(rpcStatusCode)], pipeline.metricLabels())
 }
 
-func (pipeline *AuthPipeline) metricLabels() []string {
-	labels := pipeline.AuthConfig.Labels
-	return []string{labels["namespace"], labels["name"]}
+func (pipeline *AuthPipeline) metricLabels() map[string]string {
+	labels := maps.Clone(pipeline.AuthConfig.Labels)
```

Comment:
This change causes a difference for the field name used for `AuthConfig.name`. Previous to this it would have been `authconfig`, but now it's `name` as the field name

For example: 
Before
```
auth_server_authconfig_total{authconfig="talker-api-protection",namespace="default"} 1
```
https://github.com/Kuadrant/authorino/blob/ddfd7e68f964d77ce939a45b038392708411503a/pkg/metrics/auth_config_metrics.go#L14

After:
```
auth_server_authconfig_total{name="talker-api-protection",namespace="default"} 1
```
https://github.com/Kuadrant/authorino/blob/ddfd7e68f964d77ce939a45b038392708411503a/controllers/auth_config_controller.go#L665

Is this an issue? Do we want to keep the pre-existing field name for `AuthConfig.Name`?